client/app/components/QueryControlDropdown.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Dropdown from 'antd/lib/dropdown';
+import Menu from 'antd/lib/menu';
+import Button from 'antd/lib/button';
+import Icon from 'antd/lib/icon';
+import { react2angular } from 'react2angular';
+
+
+export function QueryControlDropdown(props) {
+  const menu = (
+    <Menu>
+      {!props.query.isNew() && (!props.query.is_draft || !props.query.is_archived) && (
+        <Menu.Item>
+          <a target="_self" onClick={() => props.openAddToDashboardForm(props.selectedTab)}>
+            <Icon type="plus-circle" theme="filled" /> Add to Dashboard
+          </a>
+        </Menu.Item>
+      )}
+      {!props.query.isNew() && (
+        <Menu.Item>
+          <a onClick={() => props.showEmbedDialog(props.query, props.selectedTab)} data-test="ShowEmbedDialogButton">
+            <Icon type="share-alt" /> Embed elsewhere
+          </a>
+        </Menu.Item>
+      )}
+      {(props.queryExecuting || !props.queryResultData) && (
```

Comment:
This doesn't show up in the menu...
Cause this condition should be for a "disabled" state.